src/Snake.tsx

```diff
@@ -5,23 +5,67 @@ type Point = {
   x: number;
   y: number;
 };
+type Props = {
+  pokemonId: Pokemon['id'];
+};
 
-const GRID_SIZE = 21;
+const GRID_ROWS_SIZE = 21;
+const GRID_COLUMNS_SIZE = 25;
 
-export const SnakeStates = ({ pokemonId }: { pokemonId: Pokemon['id'] }): JSX.Element => {
+export const SnakeStates = ({ pokemonId }: Props): JSX.Element => {
+  const [speed, setSpeed] = useState(6);
   const [snakeBody, setSnakeBody] = useState<Point[]>([
-    { x: 11, y: 4 },
-    { x: 11, y: 3 },
+    { x: 4, y: 11 },
+    { x: 3, y: 11 },
   ]);
   const [food, setFood] = useState<Point>(generatePointOutSnakeBody(snakeBody));
-
+  const [direction, setDirection] = useState<Point>({ x: 1, y: 0 });
   const getElementPositionStyle = (element: Point) => {
     return {
-      gridRowStart: element.x,
-      gridColumnStart: element.y,
+      gridRowStart: element.y,
+      gridColumnStart: element.x,
     };
   };
 
+  useEffect(() => {
+    const keyPressHandler = (ev: KeyboardEvent) => {
+      switch (ev.code) {
+        case 'ArrowUp':
+          setDirection((prev) => {
+            return prev.y === 1 ? prev : { x: 0, y: -1 };
+          });
+          break;
+        case 'ArrowDown':
+          setDirection((prev) => {
+            return prev.y === -1 ? prev : { x: 0, y: 1 };
+          });
+          break;
+        case 'ArrowRight':
+          setDirection((prev) => {
+            return prev.x === -1 ? prev : { x: 1, y: 0 };
+          });
+          break;
+        case 'ArrowLeft':
+          setDirection((prev) => {
+            return prev.x === 1 ? prev : { x: -1, y: 0 };
+          });
+          break;
+      }
+    };
+    window.addEventListener('keydown', keyPressHandler);
+  }, []);
+
+  useEffect(() => {
+    const intervalId = setInterval(() => {
+      setSnakeBody((prev) => {
+        const newHeadPos: Point = outOfGameBoard({ x: prev[0].x + direction.x, y: prev[0].y + direction.y });
+        prev.length > 1 ? prev.pop() : null;
```

Comment:
Don't use the ternary operator. Use ternary operator only when it returns a value like

- `return x ? y : z`
- `setState(x ? y : z)`
- `const a = x ? y : z`

```suggestion
        if (prev.length > 1) prev.pop();
```